In [15]:
import pandas as pd
fulldata = pd.read_csv("/Users/nicolasst-germain/Documents/GitHub/audit_google_news/all_data_saisie/donneesBrutesNStG.csv")

In [16]:
fulldata

Villes Mots-clés  position du média  \
0                Longueuil  accident                  1   
1            Rouyn-Noranda  accident                  1   
2                Longueuil  accident                  2   
3            Rouyn-Noranda  accident                  2   
4            Rouyn-Noranda  accident                  3   
...                    ...       ...                ...   
488021  Dolbeau-Mistassini   déficit                  4   
488022  Dolbeau-Mistassini   déficit                  5   
488023  Dolbeau-Mistassini   déficit                  6   
488024  Dolbeau-Mistassini   déficit                  7   
488025  Dolbeau-Mistassini   déficit                  8   

                   Nom du média  \
0        Le Journal de Montréal   
1                  Radio-Canada   
2                   iHeartRadio   
3                     La Presse   
4                   iHeartRadio   
...                         ...   
488021  Le Quotidien (Saguenay)   
488022  Le Quotidien (Saguenay)   
488023  Le Quotidien (Saguenay)   
488024  Le Quotidien (Saguenay)   
488025  Le Quotidien (Saguenay)   

                                       Titre de l'article Date de publication  \
0       Longueuil: un accident de voiture force la fer...          2021-07-27   
1       Une nouvelle voie de dépassement sur la route 117          2021-07-07   
2       Accident à Longueuil: un automobiliste de 22 a...          2021-07-17   
3       Un mort et 2 blessés dans deux accidents à Rou...          2020-12-09   
4            Un motocycliste perd la vie sur la route 117          2021-07-12   
...                                                   ...                 ...   
488021  Un parc multisport de 3,3 M$ sur l'ancien terr...          2021-09-16   
488022  Dolbeau-Mistassini inaugure son café libre-ser...          2021-09-24   
488023  Dolbeau-Mistassini bonifie ses programmes d'ai...          2021-07-13   
488024  Dolbeau-Mistassini veut développer son autonom...          2021-02-15   
488025  La bureaucratie empêche une famille belge de v...          2021-10-03   

        Journée de différence  \
0                         7.0   
1                        27.0   
2                        17.0   
3                       237.0   
4                        22.0   
...                       ...   
488021                   93.0   
488022                   85.0   
488023                  158.0   
488024                  306.0   
488025                   76.0   

                                                      URL  \
0       https://www.journaldemontreal.com/2021/07/26/l...   
1       https://ici.radio-canada.ca/nouvelle/1807406/n...   
2       https://www.iheartradio.ca/energie/energie-mon...   
3       https://www.lapresse.ca/actualites/justice-et-...   
4       https://www.iheartradio.ca/energie/energie-val...   
...                                                   ...   
488021  https://www.lequotidien.com/2021/09/16/un-parc...   
488022  https://www.lequotidien.com/2021/09/24/dolbeau...   
488023  https://www.lequotidien.com/2021/07/13/dolbeau...   
488024  https://www.lequotidien.com/2021/02/16/dolbeau...   
488025  https://www.lequotidien.com/2021/10/03/la-bure...   

       date et heure saisie type mot-clés                  media_url  \
0       2021-08-03 08:00:09         local  www.journaldemontreal.com   
1       2021-08-03 08:00:09         local        ici.radio-canada.ca   
2       2021-08-03 08:00:09         local         www.iheartradio.ca   
3       2021-08-03 08:00:09         local            www.lapresse.ca   
4       2021-08-03 08:00:09         local         www.iheartradio.ca   
...                     ...           ...                        ...   
488021  2021-12-18 17:53:23      national        www.lequotidien.com   
488022  2021-12-18 17:53:23      national        www.lequotidien.com   
488023  2021-12-18 17:53:23      national        www.lequotidien.com   
488024  2021-12-18 17:53:23      national        www.lequotidien.co

# EXPLORATION DES DONNÉES GÉNÉRALES

In [17]:
fulldata["Type_media"].value_counts()

local/regional    297109
national          140304
non media          25828
hors Québec        24785
Name: Type_media, dtype: int64

Beaucoup plus de médias locaux.

In [22]:
fulldata["Type_media"].count()

488026

In [23]:
fulldata["Ville du siège social"].value_counts()

Montréal                      187573
Québec                         47695
hors Québec                    22645
Trois-Rivières                 21500
Sherbrooke                     16622
                               ...  
Kirkland                          33
Saint-Bruno-de-Montarville        20
Coteau-du-Lac                     11
Chandler                           4
Chénéville                         2
Name: Ville du siège social, Length: 75, dtype: int64

Montréal est sur-représenté dans les résultats. MAIS la-dedans tous les radio-canada sont considérés comme Montréal dans le siège social. Alors qu'il y a certainement des médias locaux là-dedans ou du moins des contenus produit localement. Certes, leurs terre d'attache est Montréal. 

In [24]:
fulldata["domaine"].value_counts()

ici.radio-canada.ca                 82580
lapresse.ca                         30034
journaldemontreal.com               29077
journaldequebec.com                 25843
lesoleil.com                        20709
                                    ...  
www.nouvellespublications.com           1
fr.blastingnews.com                     1
www.meyomessalainternational.com        1
www.cyclismactu.net                     1
www.actugaming.net                      1
Name: domaine, Length: 1444, dtype: int64

In [25]:
fulldata.count()

Villes                   488026
Mots-clés                488026
position du média        488026
Nom du média             488026
Titre de l'article       488026
Date de publication      448534
Journée de différence    448534
URL                      488026
date et heure saisie     488026
type mot-clés            488026
media_url                488026
Type_media               488026
Ville du siège social    479654
domaine                  488026
dtype: int64

In [46]:
fulldata["Villes"].value_counts()

Montréal_V2                        5200
Québec                             5189
Trois-Rivières                     5175
Montréal                           5172
Rimouski                           5148
                                   ... 
Ormstown                           3342
Boischatel                         3327
Mansfield                          3217
Chisasibi                          3134
Saint-Pierre-de-l'Île-d'Orléans    2172
Name: Villes, Length: 105, dtype: int64

^^ On voit dans ce tableau que seul Montréal_V2 compte 100% des résultats. Le plus bas est Saint-Pierre-de-l'Île-d'Orléans. Montréal_V2 est aussi le seul dont la recherche ne contenait pas le "près de ___"

# Détailler Radio-Canada et TVA

In [18]:
fulldata["domaine"].str.contains("ici.radio-canada.ca").value_counts()

False    405446
True      82580
Name: domaine, dtype: int64

In [19]:
radcan = fulldata.domaine.str.contains("ici.radio-canada.ca", na = False)
fulldata[radcan]

Villes           Mots-clés  position du média  \
1            Rouyn-Noranda            accident                  1   
9            Rouyn-Noranda            accident                  5   
11           Rouyn-Noranda            accident                  6   
15           Rouyn-Noranda            accident                  8   
17           Rouyn-Noranda            accident                  9   
...                    ...                 ...                ...   
487924           Coaticook      Biden OR Trump                  1   
487932           Coaticook             déficit                  2   
487941     Cookshire-Eaton  "premier ministre"                  2   
487998  Dolbeau-Mistassini  "premier ministre"                 17   
488010  Dolbeau-Mistassini         immigration                  7   

        Nom du média                                 Titre de l'article  \
1       Radio-Canada  Une nouvelle voie de dépassement sur la route 117   
9       Radio-Canada  Accident à Granada : les proches des victimes ...   
11      Radio-Canada        Une collision sur la route 101 fait un mort   
15      Radio-Canada  La CSN réclame une enquête sur un accident mor...   
17      Radio-Canada  Accident mortel à Granada : Mathieu Desgagné e...   
...              ...                                                ...   
487924  Radio-Canada  Pourquoi couvrir les 100 premiers jours d'un p...   
487932  Radio-Canada  Le CISSS des Îles anticipe un déficit de 4,8 M...   
487941  Radio-Canada  Fraude de 2,2 M$ : 6 ans de prison pour une an...   
487998  Radio-Canada  Deux cas de COVID-19 répertoriés dans les écol...   
488010  Radio-Canada  Le Lac-Saint-Jean, ses enjeux oubliés pendant ...   

       Date de publication  Journée de différence  \
1               2021-07-07                   27.0   
9               2021-04-30                   95.0   
11              2020-11-02                  274.0   
15              2021-06-03                   61.0   
17              2021-04-23                  102.0   
...                    ...                    ...   
487924          2021-04-26                  236.0   
487932          2021-05-27                  205.0   
487941          2021-09-24                   85.0   
487998          2021-09-13                   96.0   
488010          2021-09-19                   90.0   

                                                      URL  \
1       https://ici.radio-canada.ca/nouvelle/1807406/n...   
9       https://ici.radio-canada.ca/nouvelle/1789475/m...   
11      https://ici.radio-canada.ca/nouvelle/1746233/c...   
15      https://ici.radio-canada.ca/nouvelle/1798647/a...   
17      https://ici.radio-canada.ca/nouvelle/1787303/a...   
...                                                   ...   
487924  https://ici.radio-canada.ca/nouvelle/1787360/c...   
487932  https://ici.radio-canada.ca/nouvelle/1796589/d...   
487941  https://ici.radio-canada.ca/nouvelle/1826901/f...   
487998  https://ici.radio-canada.ca/nouvelle/1823852/p...   
488010  https://ici.radio-canada.ca/nouvelle/1825456/l...   

       date et heure saisie type mot-clés            media_url  \
1       2021-08-03 08:00:09         local  ici.radio-canada.ca   
9       2021-08-03 08:00:10         local  ici.radio-canada.ca   
11      2021-08-03 08:00:10         local  ici.radio-canada.ca   
15      2021-08-03 08:00:10         local  ici.radio-canada.ca   
17      2021-08-03 08:00:10         local  ici.radio-canada.ca   
...                     ...           ...                  ...   
487924  2021-12-18 17:49:37      national  ici.radio-canada.ca   
487932  2021-12-18 17:49:44      national  ici.radio-canada.ca   
487941  2021-12-18 17:49:51      national  ici.radio-canada.ca   
487998  2021-12-18 17:52:17      national  ici.radio-canada.ca   
488010  2021-12-18 17:52:58      national  ici.radio-canada.ca   

            Type_media Ville du siège social              domaine  
1       local/regional              Montréal  ici.radio-can

In [20]:
fulldata[radcan].Type_media.value_counts()

local/regional    65283
national          14744
hors Québec        2553
Name: Type_media, dtype: int64

Pour Rad-Can y'a bcp de contenu produit par local/régional et peu par National. Cela a été déterminé par un script produit par JHR. 

In [21]:
tva = fulldata.domaine.str.contains("tvanouvelles.ca", na = False)
fulldata[tva].Type_media.value_counts()


national          12701
local/regional     4050
Name: Type_media, dtype: int64

Quand même 4050 résultats local/régional dans TVA. 

# Création de CSV et Tableau croisée dynamique pour explorer davantage

In [48]:
## Tableau Villes par type_mots-clés 
villes_by_type_motsclés = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="type mot-clés", aggfunc="count")
villes_by_type_motsclés


type mot-clés     local  mixte  national
Villes                                  
Acton Vale         2219   1092      1106
Alma               2499   1144      1279
Amos               2537   1182      1239
Amqui              2326   1093       987
Baie-Comeau        2546   1184      1317
...                 ...    ...       ...
Val-des-Sources    2556   1170      1336
Vaudreuil-Dorion   2471   1191      1112
Victoriaville      2445   1180      1087
Ville-Marie        2587   1164      1325
Windsor            2332   1147      1261

[105 rows x 3 columns]

In [50]:
## Tableau Villes par mots-clés
villes_by_motsclés = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="Mots-clés", aggfunc="count")
villes_by_motsclés.to_csv("villes_by_motsclés.csv", na_rep="0.0")

In [26]:
## Détail du Type_media selon le type mot-clés
motcles_by_media_type = pd.pivot_table(fulldata, index="Type_media", values="domaine", columns="type mot-clés", aggfunc="count")
motcles_by_media_type

type mot-clés    local  mixte  national
Type_media                             
hors Québec      11180   4345      9260
local/regional  156658  77194     63257
national         61680  31077     47547
non media        17629   5454      2745

Ça, c'est intéressant!
On apprend que: 
- la portion média hors-québec ne compte pas pour beaucoup. 
- Pour les mots-clés locaux, mixtes et nationaux, ce sont les médias locaux qui sont les plus fréquents.
    - Cela pourrait s'expliquer par le fai que Radio-Canada compte divers chapeau et est le média le plus populaire. 
- Il y a aussi plus de résultats pour les mots-clés local que national ou mixte, mais c'est normal car plus de mots-clés à la base. 

In [14]:
## CSV qui comprend le nombre de médias le type_media par type de mot-clés par ville ##
type_media_by_ville_by_type_motsclés = pd.pivot_table(fulldata, values="domaine", aggfunc="count", index=["Villes", "type mot-clés"], columns="Type_media")
type_media_by_ville_by_type_motsclés.to_csv("villesparmotscles2.csv", na_rep="0.0")

Document généré, dispo dans le dossier data-analysis. C'est plutôt intéressant. 
On y apprend :
- Montréal_V2 est inopérable. Il manque dans les mots-clés national pour une raison incomprise à date... (l'erreur a été réparée, le fichier de Montreal_V2 national avait une nomenclature différente.) 

In [45]:
villes_by_siege_social = pd.pivot_table(fulldata, index="Villes", values="domaine", columns="Ville du siège social", aggfunc="count")
villes_by_siege_social.to_csv("villes_by_siege_social.csv", na_rep="0.0")


^^ Ce document permet de voir le siège social des médias affichés dans Google News selon la Ville de recherche. 
Par exemple, on voit qu'à Montréal, 4483 des résultats ont un siège social à Montréal. 

In [41]:
### Créer une vue où hors-québec et non media sont retirés. 
data_without_hqc = fulldata[fulldata["Type_media"].str.contains("hors Québec") == False]
data_without_hqc_nmedia = data_without_hqc[data_without_hqc["Type_media"].str.contains("non media") == False]
data_without_hqc_nmedia

Villes Mots-clés  position du média  \
0                Longueuil  accident                  1   
1            Rouyn-Noranda  accident                  1   
3            Rouyn-Noranda  accident                  2   
5                Longueuil  accident                  3   
7                Longueuil  accident                  4   
...                    ...       ...                ...   
488021  Dolbeau-Mistassini   déficit                  4   
488022  Dolbeau-Mistassini   déficit                  5   
488023  Dolbeau-Mistassini   déficit                  6   
488024  Dolbeau-Mistassini   déficit                  7   
488025  Dolbeau-Mistassini   déficit                  8   

                   Nom du média  \
0        Le Journal de Montréal   
1                  Radio-Canada   
3                     La Presse   
5            Le Courrier du Sud   
7                 TVA Nouvelles   
...                         ...   
488021  Le Quotidien (Saguenay)   
488022  Le Quotidien (Saguenay)   
488023  Le Quotidien (Saguenay)   
488024  Le Quotidien (Saguenay)   
488025  Le Quotidien (Saguenay)   

                                       Titre de l'article Date de publication  \
0       Longueuil: un accident de voiture force la fer...          2021-07-27   
1       Une nouvelle voie de dépassement sur la route 117          2021-07-07   
3       Un mort et 2 blessés dans deux accidents à Rou...          2020-12-09   
5       Un homme de 22 ans perd la vie dans un acciden...          2021-07-17   
7       Spectaculaire accident sur la route 132 en ple...          2021-06-18   
...                                                   ...                 ...   
488021  Un parc multisport de 3,3 M$ sur l'ancien terr...          2021-09-16   
488022  Dolbeau-Mistassini inaugure son café libre-ser...          2021-09-24   
488023  Dolbeau-Mistassini bonifie ses programmes d'ai...          2021-07-13   
488024  Dolbeau-Mistassini veut développer son autonom...          2021-02-15   
488025  La bureaucratie empêche une famille belge de v...          2021-10-03   

        Journée de différence  \
0                         7.0   
1                        27.0   
3                       237.0   
5                        17.0   
7                        46.0   
...                       ...   
488021                   93.0   
488022                   85.0   
488023                  158.0   
488024                  306.0   
488025                   76.0   

                                                      URL  \
0       https://www.journaldemontreal.com/2021/07/26/l...   
1       https://ici.radio-canada.ca/nouvelle/1807406/n...   
3       https://www.lapresse.ca/actualites/justice-et-...   
5       https://www.lecourrierdusud.ca/Un-homme-de-22-...   
7       https://www.tvanouvelles.ca/2021/06/18/spectac...   
...                                                   ...   
488021  https://www.lequotidien.com/2021/09/16/un-parc...   
488022  https://www.lequotidien.com/2021/09/24/dolbeau...   
488023  https://www.lequotidien.com/2021/07/13/dolbeau...   
488024  https://www.lequotidien.com/2021/02/16/dolbeau...   
488025  https://www.lequotidien.com/2021/10/03/la-bure...   

       date et heure saisie type mot-clés                  media_url  \
0       2021-08-03 08:00:09         local  www.journaldemontreal.com   
1       2021-08-03 08:00:09         local        ici.radio-canada.ca   
3       2021-08-03 08:00:09         local            www.lapresse.ca   
5       2021-08-03 08:00:09         local     www.lecourrierdusud.ca   
7       2021-08-03 08:00:09         local        www.tvanouvelles.ca   
...                     ...           ...                        ...   
488021  2021-12-18 17:53:23      national        www.lequotidien.com   
488022  2021-12-18 17:53:23      national        www.lequotidien.com   
488023  2021-12-18 17:53:23      national        www.lequotidien.com   
488024  2021-12-18 17:53:23      national        www.lequotidien.co

In [42]:
data_without_hqc_nmedia.Type_media.value_counts()

local/regional    297109
national          140304
Name: Type_media, dtype: int64